packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -177,22 +170,31 @@ def _ai_env_status_for_provider_markdown(self, provider_id):
         ):
             return na_message  # No emoji
 
+        # Avoid composing strings, to make localization easier in the future
+        not_set_title = "You have not set this environment variable, so you cannot use this provider's models."
+        set_title = "You have set this environment variable, so you can use this provider's models."
+        env_status_ok = False
         try:
-            env_var = self.providers[provider_id].auth_strategy.name
+            var_name = self.providers[provider_id].auth_strategy.name
+            env_var_display = f"`{var_name}`"
+            env_status_ok = var_name in os.environ
         except AttributeError:  # No "name" attribute
-            return na_message
-
-        output = f"`{env_var}` | "
-        if os.getenv(env_var) == None:
-            output += (
-                '<abbr title="You have not set this environment variable, '
-                + "so you cannot use this provider's models.\">❌</abbr>"
-            )
+            # Try multiple names
+            try:
+                var_names = self.providers[provider_id].auth_strategy.names
+                formatted_names = [f"`{name}`" for name in var_names]
+                env_var_display = ", ".join(formatted_names)
+                env_status_ok = all(var_name in os.environ for var_name in var_names)
+                not_set_title = "You have not set all of these environment variables, so you cannot use this provider's models."
+                set_title = "You have set all of these environment variables, so you can use this provider's models."
+            except AttributeError:  # No "names" attribute
+                return na_message
```

Comment:
Why are we trying to guess the type of this provider's `auth_strategy` with try/except blocks when we have a key that represents the type? I have two suggestions for this block:

1. Please use local variables in favor of repeating `self.providers[provider_id].auth_strategy`.
2. Please refactor this to dispatch based on the `AuthStrategy.type` field, e.g.

```py
auth_strategy = self.providers[provider_id]
if auth_strategy.type == "env":
    ...
elif auth_strategy.type == "multienv":
    ...
elif auth_strategy.type == "aws":
    ...
else:
    raise <some-exception-with-a-good-message>
```

Also, even if checking an object's attributes is the only way to determine an object's type, we should be using `hasattr()` instead of `try: ... except AttributeError: ...`.